core/src/main/scala/kafka/log/Log.scala

```diff
@@ -746,6 +753,7 @@ class Log(@volatile var dir: File,
     maybeHandleIOException(s"Exception while increasing log start offset for $topicPartition to $offset in dir ${dir.getParent}") {
       lock synchronized {
         if (offset > logStartOffset) {
+          info(s"Incrementing log start offset of partition $topicPartition to $offset in dir ${dir.getParent}")
```

Comment:
Yes, that's right. Both cases are rare enough that info-level seemed justifiable.